### PR TITLE
Feature/k8sapp

### PR DESCRIPTION
**Pull request by vsaveliev**
I will add updates of k8sapp later.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM scratch
+
+ENV {[( .EnvPrefix )]}_LOCAL_HOST 0.0.0.0
+ENV {[( .EnvPrefix )]}_LOCAL_PORT 8080
+ENV {[( .EnvPrefix )]}_LOG_LEVEL 0
+
+EXPOSE ${[( .EnvPrefix )]}_LOCAL_PORT
+
+COPY certs /etc/ssl/
```
Comment (takama) [inline Dockerfile:9]: there should be `COPY certs /etc/ssl/certs`

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+{[( .EnvPrefix )]}_LOCAL_HOST?=0.0.0.0
+{[( .EnvPrefix )]}_LOCAL_PORT?=8080
+{[( .EnvPrefix )]}_LOG_LEVEL?=0
+
+# Namespace: dev, prod, release, cte, username ...
+NAMESPACE?={[( .Namespace )]}
+
+# Infrastructure: dev, stable, test ...
+INFRASTRUCTURE?={[( .Infrastructure )]}
```
Comment (takama) [inline Makefile]: We should add KUBE_CONFIG

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+.PHONY: rm
+rm:
+ifdef HAS_EXITED
+	@echo "+ $@"
+	@docker rm ${CONTAINER_NAME}
+endif
+
+.PHONY: deploy
+deploy: push
+	helm upgrade ${CONTAINER_NAME} -f charts/${VALUES}.yaml charts --kube-context ${INFRASTRUCTURE} --namespace ${NAMESPACE} --version=${RELEASE} -i --wait
```
Comment (takama) [inline Makefile]: INFRASTRUCTURE -> KUBE_CONFIG

File: charts/values-dev.yaml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+image:
+  ## Docker Registry/Hub image name and tag
+  ##
+  registry: {[( .ChartEnvs.dev.RegistryHost )]}
+  name: {[( .ServiceName )]}
+  tag: 0.0.0
+
+  ## Docker Registry/Hub auth secret name, always use `registry-pull-secret` if registry inside if k8s
+  ##
+  # pullSecretName: registry-pull-secret
```
Comment (takama) [inline charts/values-dev.yaml]: uncomment `pullSecretName`

File: charts/values-dev.yaml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    kubernetes.io/ingress.class: nginx
+    kubernetes.io/tls-acme: 'true'
+
+  ## Ingress host feature allow access to service through host 
+  ## example: https://hostname
+  ## Must be provided if Ingress is enabled
+  ##
+  hostBased:
+    hosts:
+    - {[( .ChartEnvs.dev.ServiceHost )]}
```
Comment (takama) [inline charts/values-dev.yaml:70]: it will not work for users anyway, they have not access to DNS

File: charts/values-stable.yaml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+image:
+  ## Docker Registry/Hub image name and tag
+  ##
+  registry: {[( .ChartEnvs.stable.RegistryHost )]}
+  name: {[( .ServiceName )]}
+  tag: 0.0.0
+
+  ## Docker Registry/Hub auth secret name, always use `registry-pull-secret` if registry inside if k8s
+  ##
+  # pullSecretName: registry-pull-secret
```
Comment (takama) [inline charts/values-stable.yaml]: uncomment

File: pkg/router/bitroute/control.go [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+	}
+}
+
+// GetCode shows HTTP status code that set by Code()
+func (c *control) GetCode() int {
+	return c.code
+}
+
+// Write writes data into http output.
+func (c *control) Write(data interface{}) {
```
Comment (takama) [inline pkg/router/bitroute/control.go]: ops... old code :(

**Review by takama — CHANGES_REQUESTED** [review]

**Comment by vsaveliev** [comment]
template is updated regarding last k8sapp changes

**Pull request closed without merging**